pymc/model/transform/conditioning.py

```diff
@@ -167,11 +173,14 @@ def do(
 
     """
     do_mapping = {}
-    for var, obs in vars_to_interventions.items():
+    for var, intervention in vars_to_interventions.items():
         if isinstance(var, str):
             var = model[var]
         try:
-            do_mapping[var] = var.type.filter_variable(obs)
+            intervention = var.type.filter_variable(intervention)
+            if make_interventions_shared and isinstance(intervention, Constant):
```

Comment:
To me `make_interventions_shared` is easier than the past version, but I think that it should be more stringent. I think that if `make_interventions_shared` is `True`, it should wrap the `intervention` with a `shared` regardless of the type of data provided (be it a shared, a constant or a numpy array). Furthermore, if `make_interventions_shared` is `False` I think that the interventions should be set to `Contant`s.

Reply:
I'll have to review the code, but the thing is intervetions may not be constants, they could be graphs depending on other variables or input variables. In which case they can't be made Shared. This check for `isinstance(intervention, Constant)` was just handling that.

Reply:
Right! Ok, so then this is fine with me. But could you also handle the `make_interventions_shared=False` situation similarly? Something that checks if `intervention` is a `SharedVariable` instance and converts it to a `Constant`?

Reply:
Do we need that? Something being shared is never a problem. If the user wanted them constant he could do it themselves?